syntax = "proto3";

import "google/protobuf/struct.proto";


/**
 * UpdateIOType is a way to specify Input or Output.
 * 
 */
enum UpdateIOType {
    INPUT = 0;  /// Input Signal
    OUTPUT = 1; /// Output Signal
}

message UpdateSignal {
    UpdateIOType io = 1; /// Is this a Input or Output
    string device_type = 2; /// Is this a PWM, Digital, Analog, I2C, etc.
    google.protobuf.Value value = 3; 
}

message UpdateSignals {
    bytes guid = 1;
    map<string, UpdateSignal> signal_map = 2;
}